notifier/notifier.go

```diff
@@ -1,32 +1,36 @@
 package notifier
 
 import (
-	"bytes"
 	"context"
-	"encoding/json"
 	"errors"
 	"expvar"
 	"fmt"
 	"log"
-	"net"
 	"net/http"
 	"net/url"
 	"os"
-	"path"
 	"strings"
 	"sync"
 	"time"
 
+	"github.com/skroutz/downloader/backend"
+	httpbackend "github.com/skroutz/downloader/backend/http_backend"
+	kafkabackend "github.com/skroutz/downloader/backend/kafka_backend"
 	"github.com/skroutz/downloader/job"
 	"github.com/skroutz/downloader/stats"
 	"github.com/skroutz/downloader/storage"
 )
 
 const (
-	maxCallbackRetries = 2
-
-	statsFailedCallbacks     = "failedCallbacks"     //Counter
-	statsSuccessfulCallbacks = "successfulCallbacks" //Counter
+	maxCallbackRetries                = 2
+	statsFailedCallbacks              = "failedCallbacks"              //Counter
+	statsSuccessfulCallbacks          = "successfulCallbacks"          //Counter
+	statsUndefinedBackendWithCallback = "undefinedBackendWithCallback" //Counter
+	statsUnknownTopicOrPartition      = "unknownTopicOrPartition"      //Counter
+	// HTTP is a known backend implementation
+	HTTP = "http"
+	// KAFKA is a known backend implementation
+	KAFKA = "kafka"
```

Comment:
(from https://github.com/skroutz/downloader/pull/5#discussion_r271186399):

These constant names are too generic. Consider changing to something more specific like for example `BackendKafkaID`.

Also, the Go way (ie. standard library) is to write constants in MixedCase. For reference see:
- https://stackoverflow.com/questions/22688906/go-naming-conventions-for-const
- https://golang.org/doc/effective_go.html#mixed-caps
- standard library documentation